### DISCUSSION TITLE
Create a reusable Dropdown Menu component

### DISCUSSION THREAD

**Discussion by Guignome01** [floating-ui/floating-ui#2464] [Q&A]
I want to customize the Dropdown Menu example to make it reusable like the Popover example with the components `MenuTrigger` and `MenuContent`.

I assume that it would be used like this but I can't implement a nestable solution with this approach.

```
<Menu>
    <MenuTrigger>
        <button>Open</button>
    </MenuTrigger>
    <MenuContent>
        <MenuItem>Item 1</MenuItem>
        <Menu>
            <MenuTrigger>
                <MenuItem>Item 2</MenuItem>
            </MenuTrigger>
            <MenuContent>
                <MenuItem>Sub item 1</MenuItem>
                <MenuItem>Sub item 2</MenuItem>
            </MenuContent>
        </Menu>
        <MenuItem>Item 3</MenuItem>
    </MenuContent>
</Menu>
```

The problem is that the `<MenuItem>Item 2</MenuItem>` doesn't share the same context with the other `MenuItem` components since the global provider is in the `Menu` component so I don't have access to the `activeIndex` state.

I tried to move the provider inside the `MenuContent` component but the `MenuTrigger` component doesn't have access to the context that way and any other attempt to extract the logic into multiple hooks/providers failed.

Edit : Here is a [code sandbox](https://codesandbox.io/s/fancy-shape-gm9yvg?file=/src/App.tsx) broken demo of what I have done so far.

**Comment by cristiano-linvix** [comment]
Did you manage to do it?

**Reply by Tinkerbells** [reply]
Hello, any updates?

**Reply by cristiano-linvix** [reply]
No, i'm waiting for this too.
